scripts/generate-packages.tests.ps1

```diff
@@ -0,0 +1,132 @@
+# Unit tests for the generate-packages.ps1 script.
+# Preconditions:
+# - Install chocolatey: https://docs.chocolatey.org/en-us/choco/setup/#install-with-powershellexe
+# - Install Pester: Install-Module -Name Pester -Force -SkipPublisherCheck (https://pester.dev/docs/introduction/installation#windows)
+# Running:
+# - Open a PowerShell terminal.
+# - Generate binaries: .\scripts\its-build.ps1
+# - Package the scanner: nuget pack nuspec\netcoreglobaltool\dotnet-sonarscanner.nuspec
+# - Run the tests: Invoke-Pester -Output Detailed .\scripts\generate-packages.tests.ps1
+
+BeforeAll {
+    New-Item -Force -Path "$PSScriptRoot" -Name "testcontext" -ItemType Directory
+    New-Item -Force -Path "$PSScriptRoot/testcontext" -Name "build" -ItemType Directory
+    New-Item -Force -Path "$PSScriptRoot/testcontext/build" -Name "sonarscanner-net.zip" -type File
+    New-Item -Force -Path "$PSScriptRoot/testcontext/build" -Name "sonarscanner-net-framework.zip" -type File
+    New-Item -Force -Path "$PSScriptRoot/testcontext/scripts/version" -Name "Version.props" -type File
+
+    Set-Location "$PSScriptRoot/testcontext"
+
+    function CheckVersion([string] $packageFileName, [string] $nuspecFileName, [string] $expectedVersion) {
+        Rename-Item -Path "build/$packageFileName" -NewName 'temp.zip'
+        Expand-Archive 'build/temp.zip' -DestinationPath 'build/temp'
+
+        Get-Content -Raw "build/temp/$nuspecFileName" | Should -Match $expectedVersion
+
+        Remove-Item -Path 'build/temp.zip'
+        Remove-Item -Path 'build/temp' -Recurse
+    }
+}
+
+AfterAll {
+    Set-Location $PSScriptRoot
+    Remove-Item -Path "$PSScriptRoot/testcontext" -Recurse
+}
+
+Describe 'Main' {
+    It 'Given a release candidate version, sets short and long versions' {
```

Comment:
What do you think about using `TestCases` here?
Like this [example suggests](https://pester.dev/docs/usage/test-file-structure#more-complex-file-with-test-cases)?
It would reduce your testcases from 4 to 2, and reduce the duplication a lot.

Reply:
Nice!